lib/SPIRV/SPIRVOps.cpp

```diff
@@ -740,6 +740,102 @@ static LogicalResult verify(spirv::VariableOp varOp) {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// spv.CompositeExtractOp
+//===----------------------------------------------------------------------===//
+
+// Extracts the Result Type from Composite Type iterating all over the indices,
+// with check for out of bounds.
+static Type getElementTypeByIndices(OpAsmParser *parser, OperationState *state,
+                                    Type compositeType,
+                                    ArrayAttr indicesArrayAttr,
+                                    llvm::SMLoc attrLocation) {
+  Type resultType = compositeType;
+  for (Attribute indexAttr : indicesArrayAttr) {
+    if (!resultType) {
+      parser->emitError(attrLocation, "invalid type to extract from");
+      return nullptr;
+    }
+    // The type cast is save, because of the description in TableGen.
```

Comment:
This is not true here... The TableGen'd checks are inside `verify()` but this one is used for parsing.

Reply:
Thanks, got it, updated.